### PR TITLE
Pull out resources

### PR DESCRIPTION
**Pull request by francescaleung**
## Description

Pulls out resource structs + factory methods. Much of this is copy/pasted from the airport repo, rearranged somewhat. `ErrMissingResourceField` was added to the `resources` package (was previously in a different package); all resource structs were added to the same package. Added some checks to ensure that the structs satisfy the relevant interfaces & removed older `*ResourceFrom{Func,Env}` functions.

Part of AIR-3970.

## Test plan

Untested.

### CODE REVIEW

**Comment by francescaleung** [comment]
cc @colinking just for kicks

**Review by kellymtrinh — APPROVED** [review]

**Review by calufornia — APPROVED** [review]
thank you!

File: pkg/resources/factory_test/factory_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package factory_test
```
[the remark this answers is not in the document]
Comment (francescaleung) [reply pkg/resources/factory_test/factory_test.go:1]: yea, this is cyclic dependencies. I'll add a comment.

**Comment by francescaleung** [comment]
pulling in @colinking as a security rep

**Review by colinking — APPROVED** [review]
Sorry for the slow review -- dependency changes look fine. The reported issues are all associated with upgrades of existing dependencies and have no upgrade path.

**Pull request merged**